docs/containers-storage-architecture.md

```diff
@@ -0,0 +1,76 @@
+## containers-storage 1 "May 2024"
+
+## NAME
+containers-storage-architecture - The implementation of containers-storage 
+
+NOTE: These are implementation details subject to change. This document
+is intended to be used only as an aid for debugging and development
+of the containers-storage system. Do *not* directly access the filesystem
+components below from external code, only from the Go API.
+
+## DESCRIPTION
+The *containers/storage* library manages the following object types:
+
+- layer: A filesystem tree, along with metadata
+- image: A reference to a layer, along with metadata
+- container: A layer whose parent is an image
+
+## Layer storage
+
+### `overlay/`
+
+The `overlay/` directory contains a set of layers, named
+by their digest (note, this is the "diffid" or *uncompressed*
+digest, not the compressed one).
+
+### layer:
+
+A layer has the following components:
+
+- `diff/`: The expanded filesystem tree
+- `work/`: A directory used by `overlayfs` mounts
+- `merged/`: This will be a mount point when a layer is mounted.
+- `link`: A regular file; this contains a unique string for the layer; see `l/` below.
+- `lower`: Contains all the lower layer mounts separated by ":" and ordered from uppermost to lowermost layers.
```

Comment:
There _can_ be layers with other layout / features nowadays, but the combinatorics of `drivers/overlay` scares me.